pom.xml

```diff
@@ -944,5 +985,67 @@
         </plugins>
       </build>
     </profile>
+    <profile>
+      <activation>
+        <jdk>1.8</jdk>
+      </activation>
+      <id>checker</id>
+      <dependencies>
+        <dependency>
+          <groupId>org.checkerframework</groupId>
+          <artifactId>jdk8</artifactId>
+          <version>2.9.0</version>
+          <scope>provided</scope>
+        </dependency>
+      </dependencies>
+      <build>
+      <plugins>
+         <plugin>
+            <artifactId>maven-compiler-plugin</artifactId>
+            <version>3.6.1</version>
+            <configuration>
+              <source>1.8</source>
+              <target>1.8</target>
+              <compilerArguments>
+                <Xmaxerrs>10000</Xmaxerrs>
+                <Xmaxwarns>10000</Xmaxwarns>
+              </compilerArguments>
+              <annotationProcessorPaths>
+                <path>
+                  <groupId>org.checkerframework</groupId>
+                  <artifactId>checker</artifactId>
+                  <version>2.9.0</version>
+                </path>
+              </annotationProcessorPaths>
+              <annotationProcessors>
+                <!-- Add all the checkers you want to enable here -->
+                <annotationProcessor>org.checkerframework.checker.index.IndexChecker</annotationProcessor>
+              </annotationProcessors>
+              <compilerArgs>
+                <!-- location of the annotated JDK, which comes from a Maven dependency -->
+                <arg>-Xbootclasspath/p:${annotatedJdk}</arg>
+                <arg>-AskipDefs=/src/test/java/org/apache/commons/lang3/builder/|/src/test/java/org/apache/commons/lang3/exception/|/src/test/java/org/apache/commons/lang3/math/|/src/test/java/org/apache/commons/lang3/reflect/|/src/test/java/org/apache/commons/lang3/time/|/src/test/java/org/apache/commons/lang3/|^org\.apache\.commons\.lang3\.</arg>
```

Comment:
This configuration option is wrong. Use `-AonlyDefs=^org\.apache\.commons\.lang3\.math\.|^org\.apache\.commons\.lang3\.reflect\.|^org\.apache\.commons\.lang3\.builder\.|^org\.apache\.commons\.lang3\.time\.|^org\.apache\.commons\.lang3\.exception\.`.

To skip test files, one way to achieve that could be to surround the configuration with:
```xml
<executions>
  <execution>
    <id>default-compile</id>
    <phase>compile</phase>
    <goals>
      <goal>compile</goal>
    </goals>
    <configuration>...</configuration>
  </execution>
</executions>
```